### PR TITLE
search through tags

### PR DESCRIPTION
**Pull request by JoshuaChing**
https://github.com/CS2103JAN2017-T11-B4/main/issues/131

- include searching through tags
- not case sensitive
- change searching description to use contains rather than equals

### CODE REVIEW

File: src/main/java/seedu/task/commons/util/StringUtil.java [hunk 1]
```diff
@@ -31,7 +31,9 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         String[] wordsInPreppedSentence = preppedSentence.split("\\s+");
 
         for (String wordInSentence: wordsInPreppedSentence) {
-            if (wordInSentence.equalsIgnoreCase(preppedWord)) return true;
+            //@@joshuaching A0163673Y
+            if (wordInSentence.toLowerCase().contains(preppedWord.toLowerCase())) return true;
```
Comment (KellyLi) [inline src/main/java/seedu/task/commons/util/StringUtil.java:35]: This means that search isn't case sensitive, right? The user guide would need to be updated. : )

Comment (JoshuaChing) [reply src/main/java/seedu/task/commons/util/StringUtil.java:35]: Wasn't the search always not case sensitive?

Comment (KellyLi) [reply src/main/java/seedu/task/commons/util/StringUtil.java:35]: Search used to be case sensitive

Comment (JoshuaChing) [reply src/main/java/seedu/task/commons/util/StringUtil.java:35]: hm, I just tested on master and case isn't sensitive
wordInSentence.equalsIgnoreCase(preppedWord) <-- that ignores case

either way I've updated the docs 👍

File: src/test/java/seedu/task/logic/LogicManagerTest.java [hunk 2]
```diff
@@ -321,16 +321,16 @@ public void execute_find_invalidArgsFormat() {
...
 
     @Test
-    public void execute_find_onlyMatchesFullWordsInNames() throws Exception {
+    public void execute_find_onlyContainsWordsInDescriptions() throws Exception {
         TestDataHelper helper = new TestDataHelper();
         Task pTarget1 = helper.generatePersonWithName("bla bla KEY bla");
         Task pTarget2 = helper.generatePersonWithName("bla KEY bla bceofeia");
-        Task p1 = helper.generatePersonWithName("KE Y");
-        Task p2 = helper.generatePersonWithName("KEYKEYKEY sduauo");
+        Task p3 = helper.generatePersonWithName("KE Y");
```
Comment (KellyLi) [inline src/test/java/seedu/task/logic/LogicManagerTest.java:328]: Should this be generateTaskWithDescription?

File: src/test/java/seedu/task/logic/LogicManagerTest.java [hunk 3]
```diff
@@ -339,6 +339,33 @@ public void execute_find_onlyMatchesFullWordsInNames() throws Exception {
                 expectedList);
     }
 
+    //@@joshuaching A0163673Y
+    @Test
+    public void execute_find_containsWordsInDescriptionsAndTags() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        Task t1 = helper.generateTaskWithNameAndTags("a", "bcd", "efg", "jj");
```
Comment (KellyLi) [inline src/test/java/seedu/task/logic/LogicManagerTest.java]: Should this be generateTaskWithDescriptionAndTags?

**Review by KellyLi — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/task/logic/LogicManagerTest.java (hunk 2 above)
Comment (JoshuaChing) [reply src/test/java/seedu/task/logic/LogicManagerTest.java:328]: fixed

**Comment by JoshuaChing** [comment]
Not sure why CI is failing, DateParserTest are unrelated to my changes. Ignoring and merging.

**Pull request merged**